src/InfoViz/Native/FieldSelector/index.js

```diff
@@ -13,19 +14,48 @@ import template from './template.html';
 // ----------------------------------------------------------------------------
 
 function fieldSelector(publicAPI, model) {
+  // private variables
+  const hideField = {
+    minMax: false,
+    hist: false,
+    minMaxWidth: 0,
+    histWidth: 0,
+  };
+
+  // public API
   publicAPI.resize = () => {
     publicAPI.render();
   };
 
   publicAPI.setContainer = el => {
     if (model.container) {
       d3.select(model.container).select('div.fieldSelector').remove();
+      // Remove window listener
+      if (model.sizeSubscription) {
+        model.sizeSubscription.unsubscribe();
+        model.sizeSubscription = null;
+      }
     }
 
     model.container = el;
 
     if (el) {
+      d3.select(model.container)
+        .style('overflow-y', 'auto')
+        .style('overflow-x', 'hidden');
       d3.select(model.container).html(template);
+      model.fieldShowHistogram = model.fieldShowHistogram && (model.provider.isA('Histogram1DProvider'));
+      // append headers for histogram columns
+      if (model.fieldShowHistogram) {
+        const header = d3.select(model.container).select('thead').select('tr');
+        header.append('th').text('Min').classed(style.jsHistMin, true);
+        header.append('th').text('Histogram').classed(style.jsSparkline, true);
+        header.append('th').text('Max').classed(style.jsHistMax, true);
+      }
+      // Listen to window resize
+      model.sizeSubscription = sizeHelper.onSizeChange(publicAPI.resize);
```

Comment:
Not sure we want the component to do that. It should probably managed at the parent level (application/example).